### ISSUE TITLE
Hybrid clustering algorithm: embeddings + facet sub-grouping

### ISSUE THREAD

**Issue by paulyokota** [paulyokota/FeedForward#108]
## Summary
Implement the hybrid clustering algorithm that combines embedding-based semantic clustering with facet-based sub-grouping.

## Context
This algorithm replaces signature-based story grouping. It has two stages:

1. **Embedding clustering**: Group conversations by semantic similarity using agglomerative clustering
2. **Facet sub-grouping**: Within each embedding cluster, split by `action_type | direction` to separate directionally different issues

### Stage 1: Embedding Clustering
```python
from sklearn.cluster import AgglomerativeClustering

# Get embeddings for all conversations in the run
embeddings = np.array([conv.embedding for conv in conversations])

# Cluster using cosine distance
clustering = AgglomerativeClustering(
    n_clusters=None,
    distance_threshold=0.5,  # Tuned in prototype
    metric='cosine',
    linkage='average'
)
cluster_labels = clustering.fit_predict(embeddings)
```

### Stage 2: Facet Sub-grouping
```python
# Group by embedding cluster
cluster_groups = defaultdict(list)
for conv, label in zip(conversations, cluster_labels):
    cluster_groups[label].append(conv)

# Sub-cluster by action_type + direction
hybrid_clusters = {}
for cluster_id, convs in cluster_groups.items():
    subclusters = defaultdict(list)
    for conv in convs:
        key = f"{conv.facets.action_type} | {conv.facets.direction}"
        subclusters[key].append(conv)
    hybrid_clusters[cluster_id] = dict(subclusters)
```

### Output
Each hybrid sub-cluster becomes a candidate story group:
- Same semantic topic (embedding cluster)
- Same action type and direction (facet sub-group)

## Validated Results
Prototype on 127 conversations:
- 82 signatures → 93 embedding clusters → 109 hybrid sub-clusters
- Correctly split problematic cases:
  - "duplicate pins" (excess) vs "missing pins" (deficit)
  - "create smart pin" (creation) vs "remove smart pin" (deletion)

## Parameters
- `distance_threshold`: 0.5 (start fixed, log metrics before tuning)
- `metric`: cosine
- `linkage`: average

## Error Handling
- Conversations without embeddings or facets fall back to signature-based grouping
- Log cluster sizes and distributions for monitoring

## Dependencies
- #103 (run scoping correctness)
- #105 (data model)
- #106 (embedding generation)
- #107 (facet extraction)
- #89 (pipeline-critical tests) - need test coverage

## Prototype Reference
See `scripts/embedding_cluster_prototype.py` `cluster_conversations()` and `create_hybrid_subclusters()` functions

---

### Marcus (Sonnet) - Backend Implementation
- [ ] Create `HybridClusteringService` class
- [ ] Implement Stage 1: agglomerative clustering on embeddings
- [ ] Implement Stage 2: facet sub-grouping within clusters
- [ ] Add method to load embeddings and facets from DB for a pipeline run
- [ ] Add logging for cluster counts and size distributions

### Kenji (Sonnet) - Testing
- [ ] Write unit tests with mock embeddings/facets

### Tech Lead Review (Opus/Human)
- [ ] Design fallback strategy for conversations missing embeddings/facets
- [ ] Tune distance_threshold parameter (may need A/B testing)
- [ ] Review cluster output quality on production data
- [ ] Decide: should we expose clustering params in UI or keep fixed?
